async_processor/amqp_pub_sub.py

```diff
@@ -136,26 +133,50 @@ async def _get_channel(self) -> AbstractChannel:
         self._ch = await connection.channel()
         return self._ch
 
-    async def _get_queue(self) -> AbstractQueue:
-        if self._queue:
-            return self._queue
+    async def _validate_exchange_exists(self):
+        if not self._exchange_name:
+            return
         channel = await self._get_channel()
-        self._queue = await channel.declare_queue(self._queue_name, passive=True)
-        return self._queue
+        try:
+            # https://aio-pika.readthedocs.io/en/latest/apidoc.html#aio_pika.Channel.get_exchange
+            self._exchange = await channel.get_exchange(
+                self._exchange_name, ensure=True
+            )
+        except ChannelNotFoundEntity as ex:
```

Comment:
Is this the correct exception?

Reply:
yes, the documentation says it should be `ChannelClosed` but I was getting this instead

Reply:
```
2024-03-19 12:46:12,363:async_processor:ERROR:9362:amqp_pub_sub:_validate_exchange_exists:146 - Failed to validate exchange
Traceback (most recent call last):
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/abc.py", line 44, in __inner
    return await self.task
           ^^^^^^^^^^^^^^^
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/async_processor/amqp_pub_sub.py", line 142, in _validate_exchange_exists
    self._exchange = await channel.get_exchange(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aio_pika/channel.py", line 298, in get_exchange
    return await self.declare_exchange(name=name, passive=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aio_pika/robust_channel.py", line 181, in declare_exchange
    await super().declare_exchange(
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aio_pika/channel.py", line 271, in declare_exchange
    await exchange.declare(timeout=timeout)
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aio_pika/exchange.py", line 56, in declare
    return await channel.exchange_declare(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/channel.py", line 743, in exchange_declare
    return await self.rpc(
           ^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/base.py", line 164, in wrap
    return await self.create_task(func(self, *args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/abc.py", line 46, in __inner
    raise self._exception from e
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 684, in _wrap_awaitable
    return (yield from awaitable.__await__())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/abc.py", line 46, in __inner
    raise self._exception from e
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/channel.py", line 441, in _reader
    await hook(frame)
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/channel.py", line 410, in _on_close_frame
    raise exc
aiormq.exceptions.ChannelNotFoundEntity: NOT_FOUND - no exchange 'tezt' in vhost 'wncwyvnp'
2024-03-19 12:46:12,370:async_processor:ERROR:9362:amqp_pub_sub:_validate_exchange_exists:147 - NOT_FOUND - no exchange 'tezt' in vhost 'wncwyvnp'
Traceback (most recent call last):
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/abc.py", line 44, in __inner
    return await self.task
           ^^^^^^^^^^^^^^^
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/async_processor/amqp_pub_sub.py", line 142, in _validate_exchange_exists
    self._exchange = await channel.get_exchange(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aio_pika/channel.py", line 298, in get_exchange
    return await self.declare_exchange(name=name, passive=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aio_pika/robust_channel.py", line 181, in declare_exchange
    await super().declare_exchange(
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aio_pika/channel.py", line 271, in declare_exchange
    await exchange.declare(timeout=timeout)
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aio_pika/exchange.py", line 56, in declare
    return await channel.exchange_declare(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/channel.py", line 743, in exchange_declare
    return await self.rpc(
           ^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/base.py", line 164, in wrap
    return await self.create_task(func(self, *args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/abc.py", line 46, in __inner
    raise self._exception from e
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 684, in _wrap_awaitable
    return (yield from awaitable.__await__())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/abc.py", line 46, in __inner
    raise self._exception from e
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/channel.py", line 441, in _reader
    await hook(frame)
  File "/Users/tfy-akash/Desktop/TrueFoundry/async_service/venv/lib/python3.11/site-packages/aiormq/channel.py", line 410, in _on_close_frame
    raise exc
aiormq.exceptions.ChannelNotFoundEntity: NOT_FOUND - no exchange 'tezt' in vhost 'wncwyvnp'
```